TransformationSystem/Client/TaskManager.py

```diff
@@ -138,6 +139,71 @@ def prepareTransformationTasks( self, transBody, taskDict, owner = '', ownerGrou
         return res
       ownerDN = res['Value'][0]
 
+    try:
+      transJson = json.loads(transBody)
+      self._multiOperationsBody( transJson, taskDict, ownerDN, ownerGroup )
+    except ValueError: ##json couldn't load
+      self._singleOperationsBody( transBody, taskDict, ownerDN, ownerGroup )
+
+    return S_OK( taskDict )
+
+  def _multiOperationsBody( self, transJson, taskDict, ownerDN, ownerGroup ):
+    """ deal with a Request that has multiple operations
+
+    :param transJson: list of lists of string and dictionaries, e.g.:
+
+      .. code :: python
+
+        body = [ ( "ReplicateAndRegister", { "SourceSE":"FOO-SRM", "TargetSE":"BAR-SRM" }),
+                 ( "RemoveReplica", { "TargetSE":"FOO-SRM" } ),
+               ]
+
+    :param dict taskDict: dictionary of tasks, modified in this function
+    :param str ownerDN: certificate DN used for the requests
+    :param str onwerGroup: dirac group used for the requests
+    :returns: None
+    """
+
+    for taskID in sorted( taskDict ):
+      paramDict = taskDict[taskID]
+      if not paramDict['InputData']:
+        taskDict.pop( taskID )
+        continue
+      files = []
+
+      transID = paramDict['TransformationID']
+      oRequest = Request()
+      if isinstance( paramDict['InputData'], list ):
+        files = paramDict['InputData']
+      elif isinstance( paramDict['InputData'], basestring ):
+        files = paramDict['InputData'].split( ';' )
+
+      # create the operations from the json structure
+      for operationTuple in transJson:
+        op = Operation()
+        op.Type = operationTuple[0]
+        for parameter, value in operationTuple[1].iteritems():
+          setattr( op, parameter, value )
```

Comment:
Do you agree that only correct attributes will be here, or that I should add a check etc.?